app/discussions/[number]/comment-list.tsx

```diff
@@ -1,26 +1,52 @@
 'use client'
 
-import React from 'react'
+import React, {useState} from 'react'
+import useSWR from 'swr';
 
 import type {Comment as CommentData} from '@/app/server-utils/get-discussion-detail'
 
-export const Comment: React.FC<{comment: CommentData}> = ({comment}) => {
+const fetcher = (url: string) => fetch(url).then(r => r.json());
+
+const Comment: React.FC<{comment: CommentData, allowReplies: boolean}> = ({comment, allowReplies}) => {
+  const [viewReplies, setViewReplies] = useState(false);
+
+  const {data} = useSWR(() => (viewReplies ? `/api/comments/${comment.id}` : null), fetcher);
```

Comment:
For defense in depth, I'd `encodeURI` the comment.id. (Not crucial since it's narrowly-typed.)